lib/modules/manager/bundler/extract.ts

```diff
@@ -103,11 +104,42 @@ export async function extractPackageFile(
       });
     }
 
+    const gemGitRefsMatchRegex = regEx(
+      `^\\s*gem\\s+(['"])(?<depName>[^'"]+)['"]((\\s*,\\s*git:\\s*['"](?<gitUrl>[^'"]+)['"])|(\\s*,\\s*github:\\s*['"](?<repoName>[^'"]+)['"]))(\\s*,\\s*branch:\\s*['"](?<branchName>[^'"]+)['"])?(\\s*,\\s*ref:\\s*['"](?<refName>[^'"]+)['"])?(\\s*,\\s*tag:\\s*['"](?<tagName>[^'"]+)['"])?`,
+    );
+    const gemGitRefsMatch = gemGitRefsMatchRegex.exec(line);
+
     const gemMatchRegex = regEx(
       `^\\s*gem\\s+(['"])(?<depName>[^'"]+)(['"])(\\s*,\\s*(?<currentValue>(['"])[^'"]+['"](\\s*,\\s*['"][^'"]+['"])?))?`,
     );
     const gemMatch = gemMatchRegex.exec(line);
-    if (gemMatch) {
+
+    if (gemGitRefsMatch) {
+      const dep: PackageDependency = {
+        depName: gemGitRefsMatch.groups?.depName,
+        managerData: { lineNumber },
+      };
+      if (gemGitRefsMatch.groups?.gitUrl) {
+        const gitUrl = gemGitRefsMatch.groups.gitUrl;
+        dep.packageName = gitUrl;
+
+        if (gitUrl.startsWith('https://')) {
+          dep.sourceUrl = gitUrl.replace(/\.git$/, '');
+        }
```

Comment:
Hey, I originally found this on Flux manager ([link](https://github.com/Djiit/renovate/blob/d2b4dff8ae46c8bd7ce8bb43133286817e9a011b/lib/modules/manager/flux/extract.ts#L100-L101)).
I can get rid of this. Also, I was not sure where to use GitRefsDatasource vs GitTagsDatasource vs GithubTagsDatasource

Reply:
It can be useful to use githubtags if you know it's github, otherwise fall back to a non-github one.

Does Bundler tend to use tags-only, or also branches?

Reply:
Hey @rarkins, would you prefer I trim this from this implementation ?

Reply:
It's probably ok to leave in